kuma/users/forms.py

```diff
@@ -204,17 +204,17 @@ def save(self, request):
 
 
 class UserDeleteForm(forms.Form):
+    def __init__(self, *args, username=None, **kwargs):
+        super().__init__(*args, **kwargs)
 
-    ATTRIBUTIONS_KEEP = "keep"
-    ATTRIBUTIONS_DONATE = "donate"
-    ATTRIBUTIONS_CHOICES = (
-        (ATTRIBUTIONS_KEEP, _("Keep my attribution for my page changes")),
-        (
-            ATTRIBUTIONS_DONATE,
-            _('Switch my attribution for my page changes to "Anonymous"'),
-        ),
-    )
+        choices = {
+            "keep": _("Keep my attribution for my page changes"),
```

Comment:
```suggestion
            "keep": _("Keep my attribution for my page changes (%(username)s)") % {"username": username},
```

Reply:
I think we'll always have a truthy `username`?

Reply:
If we want to require a username, I wonder if we should enforce it by changing the constructor's signature to `def __init__(self, *args, *, username, **kwargs):`, where the `*,` indicates the beginning of keyword-only arguments. Thus, if `username` wasn't passed, we'd blow up with:

    TypeError: UserDeleteForm() missing 1 required keyword-only argument: 'username'

Or maybe that's too much new-python black magic?

Reply:
Interesting, I didn't know about that! Nice to learn something new about Python 3, thanks @callahad! Your idea sounds good to me!

Reply:
The simpler approach is to do:
```python
def __init__(self, *args, **kwargs):
    username = kwargs.pop('username')
    super().__init__(*args, **kwargs)
    ...
    ...as is...
```
that way, it's always required but doesn't have to be truthy.

Reply:
If we're going to enforce this, then I have a strong preference for `def __init__(self, *args, *, username, **kwargs)`; it does not require truthiness, it puts the args in the signature where they're more visible, and it gives a better error message: `missing 1 required keyword-only argument: 'username'` versus `KeyError: 'username'`

Reply:
Oh, cool, I misunderstood how this works. The correct syntax is just `def __init__(self, *args, username, **kwargs)`.

You only need the `*, ` if you're not otherwise collecting all the positional args.